modules/reindex/src/main/java/org/elasticsearch/index/reindex/AbstractBaseReindexRestHandler.java

```diff
@@ -43,6 +43,21 @@
                 Response extends BulkIndexByScrollResponse,
                 TA extends TransportAction<Request, Response>
             > extends BaseRestHandler {
+
+    /**
+     * @return requests_per_second from the request as a float if it was on the request, null otherwise
+     */
+    public static Float parseRequestsPerSecond(RestRequest request) {
+        String requestsPerSecond = request.param("requests_per_second");
+        if (requestsPerSecond == null) {
+            return null;
+        }
+        if ("unlimited".equals(requestsPerSecond)) {
```

Comment:
I know we sometimes use `-1` to indicate disabling something (like refresh interval), does this need to be parsed here as well?